include/alpaka/warp/WarpGenericSycl.hpp

```diff
@@ -132,6 +131,68 @@ namespace alpaka::warp::trait
             return sycl::select_from_group(actual_group, value, src);
         }
     };
+
+    template<typename TDim>
+    struct ShflUp<warp::WarpGenericSycl<TDim>>
+    {
+        template<typename T>
+        static auto shfl_up(
+            warp::WarpGenericSycl<TDim> const& warp,
+            T value,
+            std::uint32_t offset, /* must be the same for all work-items in the group */
+            std::int32_t width)
+        {
+            std::int32_t offset_int = static_cast<std::int32_t>(offset);
+            auto const actual_group = warp.m_item_warp.get_sub_group();
+            auto actual_item_id = static_cast<std::int32_t>(actual_group.get_local_linear_id());
+            auto const actual_group_id = actual_item_id / width;
+            auto const actual_src_id = actual_item_id - offset_int;
+            auto const src = actual_src_id >= actual_group_id * width
+                                 ? sycl::id<1>{static_cast<std::size_t>(actual_src_id)}
+                                 : sycl::id<1>{static_cast<std::size_t>(actual_item_id)};
+            return sycl::select_from_group(actual_group, value, src);
```

Comment:
@AuroraPerego do you think it would be worth checking if `width` is the same as `get_sub_group().get_max_local_range()`, and in that case call `shift_group_right(actual_group, value, offset)` directly, instead of `select_from_group` ?